src/WebJobs.Script.WebHost/App_Start/WebHostResolver.cs

```diff
@@ -29,13 +30,31 @@ public sealed class WebHostResolver : IDisposable
 
         private static ScriptSettingsManager _settingsManager;
 
+        private static ILoggerFactory _emptyLoggerFactory = new LoggerFactory();
+
         public WebHostResolver(ScriptSettingsManager settingsManager, ISecretManagerFactory secretManagerFactory, IScriptEventManager eventManager)
         {
             _settingsManager = settingsManager;
             _secretManagerFactory = secretManagerFactory;
             _eventManager = eventManager;
         }
 
+        public ILoggerFactory GetLoggerFactory(WebHostSettings settings)
+        {
+            WebScriptHostManager manager = GetWebScriptHostManager(settings);
+
+            if (!manager.CanInvoke())
+            {
+                // The host is still starting and the LoggerFactory cannot be used. Return
+                // an empty LoggerFactory rather than waiting. This will mostly affect
+                // calls to /admin/host/status, which do not block on CanInvoke. It allows this
+                // API to remain fast with the side effect of occassional missed logs.
+                return _emptyLoggerFactory;
```

Comment:
This seems like it could result in situations where DI injects an _emptyLoggerFactory into a valid script host instance and we lose all configured logging - how does this avoid that scenario?

Reply:
This is only used for the Web API DI -- in other words the Controllers. But yes, those controllers may now get an empty LoggerFactory. In reality, the only specific action that it should happen for is `/admin/host/status` -- as all other APIs are blocked by the CanInvoke check (https://github.com/Azure/azure-webjobs-sdk-script/blob/v1.x/src/WebJobs.Script.WebHost/Handlers/WebScriptHostHandler.cs#L50-L58)

The reality is that our dependency lifetimes stink right now -- the LoggerFactory (and TraceWriter for that matter, although it hasn't been as noticeable since we don't throw on a disposed TraceWriter) is created, configured, and disposed in the ScriptHost, yet we want access to it for logging by it's parent WebScriptHostManager. So there's this window where the LoggerFactory may be in an unusable state. This check should at least protect against that -- with the drawback that this one API (`/admin/host/status`) doesn't log to App Insights if the LoggerFactory isn't ready. It should be rare, and it's better than the current scenario where it throws. System traces still work b/c those use the TraceWriter.

@fabiocav and I have talked about this lifetime scoping in v2 and we're going to be in a much better place there with all of this. But for v1, I think this is probably the best approach for this particular issue that doesn't involve ripping apart a bunch of stuff and rebuilding it.